cmd/root.go

```diff
@@ -44,6 +45,11 @@ func Execute() {
 }
 
 func init() {
+	if os.Args[0] == docker.DOCKER_CREDS_HELPER {
+		initConfig()
+		os.Exit(docker.RunCredsHelper())
+	}
```

Comment:
It seems dangerous to me to exit from inside the `init` function (with success). In general, this behavior is undefined as there might be other `init` functions which will or will not be called in random order.
I'd recommend to move this logic into `func Execute`.

Reply:
fixed in force-push